### ISSUE TITLE
Is there a way to look at the Expected Rewards of each timestamp in the testing set?

### ISSUE THREAD

**Issue by arun-dezerv** [AI4Finance-Foundation/FinRL#1254]
For the purpose of explainability, is there a way to look at the expected rewards of each timestep of the testing set? I believe that the agent (lets say SAC) is taking an action based on the expected reward. It will be nice to see that underlying data. Please let me know if there is any way to do that

**Comment by zhumingpassional** [comment]
I do not understand you. 
the action is based on state

**Comment by arun-dezerv** [comment]
Yes. But there are probably many actions that the agent can take in every state? Why does it take a specific action? Lets say that we use a deterministic policy. The agent is probably looking at the set of all available actions and choosing the one where it expects reward to be the highest. So if I can get access to the underlying expected rewards for every state-action pair (Q table), I can understand the actions of the agent better ? Thanks a lot for replying

**Comment by arun-dezerv** [comment]
Hi. This is what I am looking for. Please see the below link

https://github.com/aleedelarica/XDRL-for-finance/blob/master/state_action_results_10000.csv

Is it possible to get such a State-Action table when using FinRL ?

**Comment by HyakunosukeOgata** [comment]
I believe he wants to express that he hopes to examine the expected rewards for all possible actions at each timestep in the test set, so that he can better understand why the agent chooses a specific action.
